editor/src/messages/tool/transform_layer/transform_layer_message_handler.rs

```diff
@@ -136,6 +214,39 @@ impl MessageHandler<TransformLayerMessage, TransformData<'_>> for TransformLayer
 				selected.original_transforms.clear();
 			}
 			TransformLayerMessage::BeginScale => {
+				if !(using_path_tool || using_select_tool) {
+					return;
+				}
+				let Some(&layer) = selected_layers.first() else {
+					return;
+				};
+				let Some(vector_data) = document.network_interface.compute_modified_vector(layer) else {
+					return;
+				};
+
+				let points = shape_editor.selected_points();
+				let selected_points: Vec<&ManipulatorPointId> = points.collect();
+
+				if using_path_tool && selected_points.is_empty() {
+					responses.add(TransformLayerMessage::CancelTransformOperation);
+					return;
+				}
+
+				if selected_points.len() == 1 {
```

Comment:
Too much nesting here too, please find ways to early return and combine `if let`s together.